cumulusci/cli/tests/test_config.py

```diff
@@ -34,33 +37,62 @@ def test_init(self):
             self.assertIn(key, config.keychain.config)
         self.assertIn(config.project_config.repo_root, sys.path)
 
-    @mock.patch("cumulusci.cli.config.CliConfig._load_project_config")
+    @mock.patch("cumulusci.cli.config.CliRuntime._load_project_config")
     def test_load_project_not_in_project(self, load_proj_cfg_mock):
         load_proj_cfg_mock.side_effect = NotInProject
 
         with self.assertRaises(click.UsageError):
-            CliConfig()
+            CliRuntime()
 
-    @mock.patch("cumulusci.cli.config.CliConfig._load_project_config")
+    @mock.patch("cumulusci.cli.config.CliRuntime._load_project_config")
     def test_load_project_config_no_file(self, load_proj_cfg_mock):
         load_proj_cfg_mock.side_effect = ProjectConfigNotFound
         with self.assertRaises(click.UsageError):
-            CliConfig()
+            CliRuntime()
 
-    @mock.patch("cumulusci.cli.config.CliConfig._load_project_config")
+    @mock.patch("cumulusci.cli.config.CliRuntime._load_project_config")
     def test_load_project_config_error(self, load_proj_cfg_mock):
         load_proj_cfg_mock.side_effect = ConfigError
 
         with self.assertRaises(click.UsageError):
-            CliConfig()
+            CliRuntime()
+
+    @mock.patch("cumulusci.cli.config.keyring")
+    def test_get_keychain_key__migrates_from_env_to_keyring(self, keyring):
+        keyring.get_password.return_value = None
+
+        config = CliRuntime()
+        self.assertEqual(self.key, config.keychain.key)
+        keyring.set_password.assert_called_once_with(
+            "cumulusci", "CUMULUSCI_KEY", self.key
+        )
+
+    @mock.patch("cumulusci.cli.config.keyring")
+    def test_get_keychain_key__env_takes_precedence(self, keyring):
+        keyring.get_password.return_value = "overridden"
 
-    def test_load_keychain__no_key(self):
-        with mock.patch.dict(os.environ, {"CUMULUSCI_KEY": ""}):
-            with self.assertRaises(click.UsageError):
-                config = CliConfig()
+        config = CliRuntime()
+        self.assertEqual(self.key, config.keychain.key)
+
+    @mock.patch("cumulusci.cli.config.keyring")
+    def test_get_keychain_key__generates_key(self, keyring):
+        del os.environ["CUMULUSCI_KEY"]
+        keyring.get_password.return_value = None
+
+        config = CliRuntime()
+        self.assertNotEqual(self.key, config.keychain.key)
+        self.assertEqual(16, len(config.keychain.key))
+
+    @mock.patch("cumulusci.cli.config.keyring")
+    def test_get_keychain_key__warns_if_generated_key_cannot_be_stored(self, keyring):
```

Comment:
I hadn't noticed this dunder in the function anywhere else. I think it's fine, just curious if it has some different behavior or is purely aesthetic?

Reply:
I tend to follow a pattern of test_[method name]__[short description of variation being tested] -- the dunder is just to make it clearer which part is the method name.